clang/test/SemaTemplate/GH55509.cpp

```diff
@@ -0,0 +1,84 @@
+// RUN: %clang_cc1 -fsyntax-only -verify -std=c++26 %s
```

Comment:
nit: we might want to move it to an existing file (so we don't have tests scattered around), but it's up to you if you feel the test is long enough to have a new file.

Reply:
Yeah I think this is big enough to be on its own file.